api/oas3.0/v2/device-sdk.yaml

```diff
@@ -0,0 +1,1269 @@
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '500':
+          description: The device driver is unable to process the request, or too many values were returned.
+          headers:
+            X-Correlation-ID:
+              $ref: '#/components/headers/correlatedResponseHeader'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+    put:
+      description: Request the actuator by its id to trigger a action or set a current value for the command or device resource specified.
+      parameters:
+        - $ref: '#/components/parameters/correlatedRequestHeader'
+        - in: path
+          name: id
+          required: true
+          schema:
+            type: string
+          example: b4081782-a611-4180-848a-cf8235e6351d
+        - in: path
+          name: command
+          required: true
+          schema:
+            type: string
+          example: allValues
+      responses:
+        '200':
+          description: The PUT command was successful.
+          headers:
+            X-Correlation-ID:
+              $ref: '#/components/headers/correlatedResponseHeader'
+        '404':
+          description: If no device exists for the ID provided or the command is unknown.
+          headers:
+            X-Correlation-ID:
+              $ref: '#/components/headers/correlatedResponseHeader'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '405':
+          description: If the requested command exists but not for PUT, or the resource is marked as read-only.
+          headers:
+            X-Correlation-ID:
+              $ref: '#/components/headers/correlatedResponseHeader'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '423':
+          description: If the device or service is locked (admin state) or disabled (operating state).
+          headers:
+            X-Correlation-ID:
+              $ref: '#/components/headers/correlatedResponseHeader'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '500':
+          description: The device driver is unable to process the request.
+          headers:
+            X-Correlation-ID:
+              $ref: '#/components/headers/correlatedResponseHeader'
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SettingRequest'
+        required: true
+
+  /discovery:
+    post:
+      description: Run the discovery request for a Device Service.
+      responses:
```

Comment:
RequestBody? i.e. for consistency we should have a request DTO for every API? If not, then document that not request body required.

Reply:
Noted no request body